### PR TITLE
Create CHANGELOG.md to contain release information

### LINKED ISSUE #178: Create CHANGELOG.md to contain release information
Create a CHANGELOG.md file (based roughly on Argo CD's) in order to contain release information. 

The same content will also be used in the Releases panel, for the release tag.

### PR DESCRIPTION
**Pull request by jgwest**
Fixes  #178

### CODE REVIEW

**Review by wtam2018 — APPROVED** [review]
LGTM!
Thanks @jgwest

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+BUT, best of all, all these features are controlled by only a single ApplicationSet Kubernetes custom resource, which means no more juggling of multiple Argo CD Application resources to target those multiple clusters/repos!
+
+Unlike with an Argo CD Application resource, which deploys resources from a single Git repository to a single destination cluster/namespace, ApplicationSet uses templated automation to create, modify, and manage multiple Argo CD applications at once.
+
+You can learn more about from the [ApplicationSet documentation](https://argocd-applicationset.readthedocs.io) or check out the source [and learn how you can contribute](https://github.com/argoproj-labs/applicationset/).
+
+Since this is our first release, we would ❤️ if you would give ApplicationSets a shot, and let us know what cool problems you are solving with it, or what pain points you hit.
+Got feature requests, bug reports, or want to contribute code? Let us know on our project repository, or join us on #argo-cd-appset on Slack.
```
Comment (OmerKahani) [inline CHANGELOG.md:19]:
```suggestion
Got feature requests, bug reports, or want to contribute code? Let us know on our project repository, or join us on [#argo-cd-appset](https://argoproj.slack.com/archives/C014ZPM32LU) on Slack.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
+# Changelog
+
+# v0.1.0
+
+I am excited to announce the first release of the Argo CD ApplicationSet controller, v0.1.0, releasing now alongside Argo CD v2.0!
+
+The ApplicationSet controller provides the ability:
+- To deploy Argo CD Applications to multiple Kubernetes cluster at once
```
Comment (OmerKahani) [inline CHANGELOG.md:8]:
```suggestion
- To deploy Argo CD Applications to multiple Kubernetes clusters at once
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+- To deploy multiple Argo CD applications from a single monorepo
+- Allows unprivileged cluster users (those without access to the Argo CD namespace) to deploy Argo CD applications without the need to involve cluster administrators in enabling the destination clusters/namespaces
+
+BUT, best of all, all these features are controlled by only a single ApplicationSet Kubernetes custom resource, which means no more juggling of multiple Argo CD Application resources to target those multiple clusters/repos!
+
+Unlike with an Argo CD Application resource, which deploys resources from a single Git repository to a single destination cluster/namespace, ApplicationSet uses templated automation to create, modify, and manage multiple Argo CD applications at once.
+
+You can learn more about from the [ApplicationSet documentation](https://argocd-applicationset.readthedocs.io) or check out the source [and learn how you can contribute](https://github.com/argoproj-labs/applicationset/).
```
Comment (OmerKahani) [inline CHANGELOG.md:16]:
```suggestion
You can learn more about this from the [ApplicationSet documentation](https://argocd-applicationset.readthedocs.io) or check out the source [and learn how you can contribute](https://github.com/argoproj-labs/applicationset/).
```

File: CHANGELOG.md (hunk 1 above)
Comment (jgwest) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 3 above)
Comment (jgwest) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (jgwest) [reply CHANGELOG.md]: Done.

**Review by OmerKahani — APPROVED** [review]
exciting

**Pull request merged**